pkg/controller/vsphere/reconciler.go

```diff
@@ -1241,6 +1234,22 @@ type attachedDisk struct {
 	diskMode string
 }
 
+// Filters out disks that look like vm OS disk.
+// VM os disks filename contains the machine name in it
+// and has the format like "[DATASTORE] path-within-datastore/machine-name.vmdk".
+// This is based on vSphere behavior, an OS disk file gets a name that equals the target VM name during the clone operation.
+func filterOutVmOsDisk(attachedDisks []attachedDisk, machine *machinev1.Machine) []attachedDisk {
+	var disks []attachedDisk
+
+	for _, disk := range attachedDisks {
+		if strings.HasSuffix(disk.fileName, fmt.Sprintf("%s.vmdk", machine.GetName())) {
```

Comment:
Rather than suffix, could we split on `/` and take the last entry and compare for equality? Would that be more deterministic?

Reply:
>Would that be more deterministic?

I don't feel that it would be more deterministic, to be honest.
One potential issue comes to my mind - in case vmdk will lay down in a datastore root filename might have a format like `[DATASTORE] foo.vmdk`. Not sure if it's a valid case though.

Reply:
Can we ask the splat team, they might know?

Reply:
Discussed here: https://redhat-internal.slack.com/archives/CH06KMDRV/p1675430846007109

I updated this suffix to include a starting slash in there, which should be equivalent to what you proposed above. Extended tests for this also.